framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -264,17 +282,25 @@ public void updateFromFieldAssignment(
             return;
         }
 
-        String className = getEnclosingClassName(lhs.getElement());
+        ClassSymbol enclosingClass = ((VarSymbol) lhs.getElement()).enclClass();
+
+        // do not infer types for code that isn't presented as source
+        if (!ElementUtils.isElementFromSourceCode(enclosingClass)) {
+            return;
+        }
+
+        @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+        @DotSeparatedIdentifiers String className = enclosingClass.flatname.toString();
```

Comment:
The documentation of `ClassSymbol.flatname` is:

```
        /** the fully qualified name of the class after converting to flat
         *  representation, i.e. pck.outer$inner,
         *  set externally for local and anonymous classes
         */
        public Name flatname;
```

so I think the checker warnings arises not from #3094 but from the fact that the annotation is wrong and should be `@BinaryName`.  Another reason to think that it's not #3094 is that `javac.astub` contains no comment about the issue.  Please also examine other mentions of #3094 in the diffs.

Reply:
This referenced #3094 because of the call to `Name#toString`, which cannot be referenced in the stub file. `javac.astub` does, specifically, mention #3094 in a comment to justify why `Name#toString` is *not* annotated.

That said, I think you're right about this particular case being `@BinaryName`, although the warning will still be issued and will still need to be suppressed. I'll check the other suppressions related to #3094 for similar mistakes, too.